PROTO_tests/tests/io_test.py

```diff
@@ -0,0 +1,413 @@
+        # compression doesn't work for bool see issue #2579
+        ak_arr = make_ak_arrays(prob_size * pytest.nl, dtype)
+
+        with tempfile.TemporaryDirectory(dir=TestParquet.par_test_base_tmp) as tmp_dirname:
+            ak_arr.to_parquet(
+                f"{tmp_dirname}/pq_testcorrect", "my-dset", compression=comp if dtype != "bool" else None
+            )
+            pq_arr = ak.read_parquet(f"{tmp_dirname}/pq_testcorrect*", "my-dset")
+            assert (ak_arr == pq_arr).all()
+
+            # verify generic read works
+            gen_arr = ak.read(f"{tmp_dirname}/pq_testcorrect*", "my-dset")
+            assert (ak_arr == gen_arr).all()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NUMERIC_AND_STR_TYPES)
+    def test_multi_file(self, prob_size, dtype):
+        is_multi_loc = pytest.nl != 1
+        NUM_FILES = pytest.nl if is_multi_loc else 2
+        adjusted_size = int(prob_size / NUM_FILES) * NUM_FILES
+        ak_arr = make_ak_arrays(adjusted_size, dtype)
+
+        per_arr = int(adjusted_size / NUM_FILES)
+        with tempfile.TemporaryDirectory(dir=TestParquet.par_test_base_tmp) as tmp_dirname:
+            file_name = f"{tmp_dirname}/pq_test"
+            if is_multi_loc:
+                # when multi locale multiple created automatically
+                ak_arr.to_parquet(file_name, "test-dset")
+            else:
+                # when single locale artifically create multiple files
+                for i in range(NUM_FILES):
+                    arr_in_file_i = ak_arr[(i * per_arr) : (i * per_arr) + per_arr]
+                    arr_in_file_i.to_parquet(f"{file_name}{i:04d}", "test-dset")
+
+            assert len(glob.glob(f"{file_name}*")) == NUM_FILES
+            pq_arr = ak.read_parquet(f"{file_name}*", "test-dset")
+            assert (ak_arr == pq_arr).all()
+
+    def test_wrong_dset_name(self):
+        ak_arr = ak.randint(0, 2**32, 100)
+        with tempfile.TemporaryDirectory(dir=TestParquet.par_test_base_tmp) as tmp_dirname:
+            ak_arr.to_parquet(f"{tmp_dirname}/pq_test", "test-dset-name")
+
+            with pytest.raises(RuntimeError):
+                ak.read_parquet(f"{tmp_dirname}/pq_test*", "wrong-dset-name")
+
+            with pytest.raises(ValueError):
+                ak.read_parquet(f"{tmp_dirname}/pq_test*", ["test-dset-name", "wrong-dset-name"])
+
+    @pytest.mark.parametrize("dtype", NUMERIC_AND_STR_TYPES)
+    @pytest.mark.parametrize("comp", COMPRESSIONS)
+    def test_edge_case_read_write(self, dtype, comp):
+        np_edge_case = make_edge_case_arrays(dtype)
+        ak_edge_case = ak.array(np_edge_case)
+        with tempfile.TemporaryDirectory(dir=TestParquet.par_test_base_tmp) as tmp_dirname:
+            ak_edge_case.to_parquet(
+                f"{tmp_dirname}/pq_test_edge_case",
+                "my-dset",
+                compression=comp if dtype != "bool" else None,
+            )
```

Comment:
Can we add another note about the compression on bools here? I just don't want this to be missed down the road.